client/components/marketing-survey/cancel-jetpack-form/index.tsx

```diff
@@ -193,11 +206,16 @@ const CancelJetpackForm: React.FC< Props > = ( {
 
 	const handleCloseDialog = () => {
 		props.onClose();
-		resetSurveyState();
-
 		// record tracks event
 		// sends the same event name as the main product cancellation form
 		recordEvent( 'calypso_purchases_cancel_form_close' );
+
+		// When an offer has been accepted, redirect back to the purchases page.
+		if ( cancellationOfferApplySuccess || steps.OFFER_ACCEPTED_STEP === cancellationStep ) {
+			page.redirect( props.purchaseListUrl );
```

Comment:
Should we make sure the URL is not external, here?

Reply:
Couldn't hurt to add that check, thanks!